lib/synapse/haproxy.rb

```diff
@@ -655,18 +693,33 @@ def generate_frontend_stanza(watcher, config)
     end
 
     def generate_backend_stanza(watcher, config)
+      backends = {}
+
+      # The ordering here is important.  First we add all the backends in the
+      # disabled state...
+      @seen.fetch(watcher.name, []).each do |backend_name, backend|
+        backends[backend_name] = backend.merge('enabled' => false)
+      end
+
+      # ... and then we overwite any backends that the watchers know about,
+      # setting the enabled state.
+      watcher.backends.each do |backend|
+        backend_name = construct_name(backend)
+        backends[backend_name] = backend.merge('enabled' => true)
+      end
+
       if watcher.backends.empty?
-        log.warn "synapse: no backends found for watcher #{watcher.name}"
+        log.debug "synapse: no backends found for watcher #{watcher.name}"
       end
 
       stanza = [
         "\nbackend #{watcher.name}",
         config.map {|c| "\t#{c}"},
-        watcher.backends.shuffle.map {|backend|
-          backend_name = construct_name(backend)
+        backends.map {|backend_name, backend|
```

Comment:
I think the shuffling is helpful for some of the things that we have that have a predictable number (per box) of long lived connections like our MySQL databases.  Especially since these connections survive haproxy restarts and the time of reconnection is very synchronized on every server, I could definitely see this causing problems for our use case.

This may help with #127, but unless I'm missing something, I don't understand how consistent hashing could be expected to play nicely with synapse/haproxy.